### ISSUE TITLE
Strict equal error not caught in strict mode

### ISSUE THREAD

**Issue by ain** [eslint/eslint#6537]
**What version of ESLint are you using?**

`2.13.1`

**What parser (default, Babel-ESLint, etc.) are you using?**

`default`

**Please show your full configuration:**

```json
{
  "env": {
    "browser": true,
    "commonjs": true,
    "es6": true
  },
  "extends": "eslint:recommended",
  "parserOptions": {
    "ecmaVersion": 6,
    "sourceType": "module",
    "ecmaFeatures": {
      "impliedStrict": true
    }
  },
  "rules": {
    "indent": [
      "error",
      2
    ],
    "linebreak-style": [
      "error",
      "unix"
    ],
    "quotes": [
      "error",
      "single"
    ],
    "semi": [
      "error",
      "always"
    ]
  }
}
```
**What did you do? Please include the actual source code causing the issue.**

1. Create script `"eslint": "eslint src/*.js"`
2. Create JS file at `src/` with 

  ```js
if (something == somethingElse) {
  ...
}
  ```
3. Run `npm run eslint`

**What did you expect to happen?**

Get strict equal error in respect to strict mode.

**What actually happened? Please include the actual, raw output from ESLint.**

Linting passed.

**Comment by michaelficarra** [comment]
Which rule, exactly, were you expecting to fail?

**Comment by platinumazure** [comment]
@ain Just to be clear: Do you believe that if a script or section of code has `"use strict";` in it, that a loose equality check (`a == b`) should show up as a lint error?

Everything I have ever read about strict mode still allows loose equality checks.

If you want to have lint errors show up for loose equality checks, the rule you want is [`eqeqeq`](http://eslint.org/docs/rules/eqeqeq).

**Comment by ain** [comment]
Excellent! Thanks a lot for your support @platinumazure!
